Support 3.3.x (x != 0)

znz opened the pull request:
## Why?

`<= 3.3` means `<= 3.3.0`, so this cannot install with 3.3.1 or later.

```
scimaenaga-1.0.2 requires ruby version >= 2.5.9, <= 3.3, which is incompatible
with the current version, ruby 3.3.3p89
```

## What?

`< 3.4` supports 3.3.x (x != 0) too.

## Caveats

Nothing affects already supported versions.

znz committed:
Support 3.3.x (x != 0)

valerauko commented on scimaenaga.gemspec:
```diff
@@ -15,7 +15,7 @@ Gem::Specification.new do |s|
 
   s.files = Dir['{app,config,db,lib}/**/*', 'MIT-LICENSE', 'Rakefile', 'README.md']
 
-  s.required_ruby_version = '>= 2.5.9', '<= 3.3'
```
Please add your problematic version (3.3.3?) to the test matrix after 3.3.0 at https://github.com/StudistCorporation/scimaenaga/blob/5d90e9e99a65ed60d457470ad510b51374ce3fc0/.github/workflows/test.yaml#L58

valerauko reviewed (changes requested):
Thanks for the contribution!

mtakeda15 commented:
@znz 
I want to keep updating test.yml when new ruby version is released, so I prefer `<= 3.3.3` to `<3.4`.

znz committed:
Update ruby 3.3 from 3.3.0 to 3.3.3

https://www.ruby-lang.org/en/news/2024/06/12/ruby-3-3-3-released/

znz committed:
Change from `< 3.4` to `<= 3.3.3`

https://github.com/StudistCorporation/scimaenaga/pull/52#issuecomment-2165076579

mtakeda15 reviewed (approved):
Thank you for your contribution!!
